src/main/java/com/conveyal/r5/transit/TransportNetworkCache.java

```diff
@@ -374,12 +405,50 @@ private TransportNetwork buildNetworkFromManifest (String networkId) {
         return network;
     }
 
+    private String getManifestKey(String networkId) {
+        String filename = getManifestFilename(networkId);
+        return bucketFolder != null ? String.join("/", bucketFolder, filename) : filename;
+    }
+
+    private String getManifestFilename(String networkId) {
+        return GTFSCache.cleanId(networkId) + ".json";
+    }
+
+    private LoadingCache<String, TransportNetwork> createCache(int size) {
+        RemovalListener<String, TransportNetwork> removalListener = removalNotification -> {
+            String id = removalNotification.getKey();
+
+            // delete local files ONLY if using s3
+            if (sourceBucket != null) {
+                String[] extensions = {".db", ".db.p", ".zip"};
+                // delete local cache files (including zip) when feed removed from cache
+                for (String type : extensions) {
+                    File file = new File(cacheDir, id + type);
+                    file.delete();
+                }
+            }
+        };
+        return CacheBuilder.newBuilder()
+                .maximumSize(size)
+                .removalListener(removalListener)
+                .build(new CacheLoader() {
```

Comment:
Oh, right, I tried putting a lambda here and java is too stupid to figure out the types.